res/controllers/Vestax-VCI-100MKII-scripts.js

```diff
@@ -257,32 +257,53 @@ VCI102.next_effect = function(ch, midino, value, status, group) {
     }
 };
 
-VCI102.loopSize = [4, 4, 4, 4];  // saved beatloop_size
-
 VCI102.loop = function(ch, midino, value, status, group) {
     if (value) {
         if (engine.getValue(group, "loop_enabled")) {
             engine.setValue(group, "reloop_toggle", 1);
-            // restore beatloop_size
-            engine.setValue(group, "beatloop_size", VCI102.loopSize[ch]);
         } else {
-            VCI102.loopSize[ch] = engine.getValue(group, "beatloop_size");
             engine.setValue(group, "beatloop_activate", 1);
+            engine.setValue(group, "beatloop_activate", 0);
         }
     }
 };
 
 VCI102.reloop = function(ch, midino, value, status, group) {
     if (value) {
         if (engine.getValue(group, "loop_enabled")) {
+            engine.setValue(group, "loop_out", 1);
             engine.setValue(group, "loop_out", 0);
         } else {
-            VCI102.loopSize[ch] = engine.getValue(group, "beatloop_size");
             engine.setValue(group, "reloop_toggle", 1);
         }
     }
 };
 
+VCI102.loop_scale = function(ch, group, scale) {
+    if (VCI102.shift[1 - ch % 2]) {
+        // scale beatjump_size if shift of the other Deck
+        engine.setValue(group, "beatjump_size",
+                        engine.getValue(group, "beatjump_size") * scale);
+    } else if (engine.getValue(group, "loop_enabled")) {
+        engine.setValue(group, "loop_scale", scale);
```

Comment:
loop_scale is only for manual loops set with loop_in and loop_out which are not mapped on this controller.